src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java

```diff
@@ -126,48 +124,52 @@ public void handle(Endpoint<?> endpoint, Handler<?> handler) {
         }
         List<HttpBridgeResult<?>> results = new ArrayList<>(records.size());
 
-        // start sending records asynchronously
-        if (isAsync) {
-            // if async is specified, return immediately once records are sent
-            for (KafkaProducerRecord<K, V> record : records) {
-                this.send(record, null);
+        // fulfilling the request of sending (multiple) record(s) sequentially but in a separate thread
+        // this will free the Vert.x event loop still in place
+        CompletableFuture.runAsync(() -> {
```

Comment:
Do we know whether the common fork-join pool is appropriate for what we're doing here? The Javadoc for [ForkJoinPool](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ForkJoinPool.html) has some things to say about unmanaged blocking (and `Producer.send()` does block) so it's not immediately obvious to me whether it's appropriate here or not.

Reply:
I have thought about this and my intention was to come back to this later, starting to use the forkjoin pool for now. Do you see it something to address right now?

Reply:
I think we should have an idea of what the ultimate destination is at this point, even if we plan to get there is steps. But right it's not clear to me that we do know an ultimate destination.

Reply:
My plan was somehow ... "let's try to mimic the Vert.x behaviour while still having Vert.x in the picture", I mean ...
You enter the `HttpSourceBridgeEndpoint.handle` method through the Vert.x event loop thread. Then, in the Vert.x based case, when using the Vert.x Kafka client it's going to use Vert.x worker threads to run Kafka stuff and then back to the event loop for the handlers.
The idea was, entering as Vert.x event loop thread and freeing it by running the Kafka stuff through CompletableFuture asynchronously (using the forkjoin thread pool for now). What we cannot guarantee is the way back on the Vert.x event loop thread.